### ISSUE TITLE
Determine requirements

### ISSUE THREAD

**Issue by penguinyaro** [HACC2022/SquidBrainsAreShapedLikeDonuts#1]
Challenge prompt: [HACC-2022-NIC_Data-Visualization-Challenge.pdf](https://github.com/HACC2022/SquidBrainsAreShapedLikeDonuts/files/9806620/HACC-2022-NIC_Data-Visualization-Challenge.pdf)

> Agencies gather data through everyday operations. Some of this data is available online via https://opendata.hawaii.gov. Often agencies need to present and explain data to other agencies, the legislature, and the public.There’s a need for easily turning data into visualizations and at no cost.
>
> **Goal**: Provide an online tool to streamline the process of displaying data from https://opendata.hawaii.gov/dataset by providing dynamic data visualizations for users of the Opendata site

Based on the above document and what we have learned about the project from NIC and their presentation, what are the needs of the customer?

From the needs, let us develop the requirements for our solution.

**Comment by penguinyaro** [comment]
- Pulls data from https://opendata.hawaii.gov/
  - Is a separate site (i.e. is not integrated within opendata website)

**Comment by penguinyaro** [comment]
- Transforms the text into a more easily digestible forms that can be easily digested by the user

**Comment by hmurai8** [comment]
- Separate website that can also be downloaded as an image file

**Comment by hmurai8** [comment]
- On the website, takes in user's linked data through a few clicks
- Generates data visualization with labels, but with the option for the user to customize

**Comment by penguinyaro** [comment]
_free_

There will be some cost associated with hosting the site somewhere, but we shouldn't use any tools that add an additional cost (e.g. license fees)

**Comment by hmurai8** [comment]
- Easily accessible to agency staff

**Comment by hmurai8** [comment]
1. User pastes dataset url into website program (unless pull api) (**flask**)
2. Program reads, cleans, and aggregates data (**pandas**)
3. Program generates pivot table (or something similar) for user to see (**mito? pandas?**)
4. Program provides guidance on how to customize the pivot table (**flask**)
5. Program provides user the option of which type of data visualization to generate (**flask, matplotlib**)
6. Program gives some examples of other datasets visualization?? (**matplotlib**)
7. User chooses and clicks to customize and generate visualizations

**Comment by penguinyaro** [comment]
> 1. User pastes dataset url into website program (unless pull api) (**flask**)
> 2. Program reads, cleans, and aggregates data (**pandas**)
> 3. Program generates pivot table (or something similar) for user to see (**mito? pandas?**)
> 4. Program provides guidance on how to customize the pivot table (**flask**)
> 5. Program provides user the option of which type of data visualization to generate (**flask, matplotlib**)
> 6. Program gives some examples of other datasets visualization?? (**matplotlib**)
> 7. User chooses and clicks to customize and generate visualizations

I agree this is the workflow we want to strive for. Do want to know that these are more of a _design specifications_ than _requirements_ though. 

We should organize a master list for requirements as well. I think it is one of the final things we need to fill out. The things you mentioned earlier were good. They are things that we can judge our end result by, if we meet them, then we're good.
